ext/standard/uuid.c

```diff
@@ -0,0 +1,662 @@
+		return;
+	}
+
+	nsid = get_uuid(namespace);
+	if (nsid == NULL) {
+		return;
+	}
+
+	php_uuid_create_v5(&uuid, nsid, Z_STRVAL_P(name), Z_STRLEN_P(name));
+	new_uuid(return_value, &uuid);
+}
+ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX(UUID_v5_args, 0, 2, self, 0)
+ZEND_ARG_OBJ_INFO(0, namespace, self, 0)
+ZEND_ARG_TYPE_INFO(0, name, IS_STRING, 0)
+ZEND_END_ARG_INFO()
+
+#define PHP_UUID_NAMED_CONSTRUCTOR(php_method_name, c_function_name)            \
+	PHP_METHOD(UUID, php_method_name)                                           \
+	{                                                                           \
+		const php_uuid c_function_name = php_uuid_##c_function_name();          \
+		                                                                        \
+		if (zend_parse_parameters_none_throw() == FAILURE) {                    \
+			return;                                                             \
+		}                                                                       \
+		                                                                        \
+		new_uuid(return_value, &c_function_name);                               \
+	}                                                                           \
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO(UUID_##php_method_name##_args, self, 0) \
+	ZEND_END_ARG_INFO();
+
+PHP_UUID_NAMED_CONSTRUCTOR(NamespaceDNS, namespace_dns)
+PHP_UUID_NAMED_CONSTRUCTOR(NamespaceOID, namespace_oid)
+PHP_UUID_NAMED_CONSTRUCTOR(NamespaceURL, namespace_url)
+PHP_UUID_NAMED_CONSTRUCTOR(NamespaceX500, namespace_x500)
+PHP_UUID_NAMED_CONSTRUCTOR(Nil, nil)
+
+PHP_METHOD(UUID, __clone)
+{
+	zend_throw_error(zend_ce_error, "Cannot clone immutable %s object", ZSTR_VAL(php_ce_UUID->name));
+}
+ZEND_BEGIN_ARG_INFO(UUID___clone_args, NULL)
+ZEND_END_ARG_INFO()
+
+PHP_METHOD(UUID, __set)
+{
+	zend_throw_error(zend_ce_error, "Cannot set dynamic properties on immutable %s object", ZSTR_VAL(php_ce_UUID->name));
+}
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUID___set_args, IS_VOID, 0)
+	ZEND_ARG_INFO(0, _)
+	ZEND_ARG_INFO(0, __)
+ZEND_END_ARG_INFO()
+
+PHP_METHOD(UUID, __wakeup)
+{
+	zval *bytes = zend_read_property(php_ce_UUID, &EX(This), UUID_BYTES_PROP, UUID_BYTES_PROP_LEN, 1, NULL);
+
+	if (zend_parse_parameters_none_throw() == FAILURE) {
+		return;
+	}
+
+	if (Z_TYPE_P(bytes) != IS_STRING) {
+		zend_throw_exception_ex(
+			spl_ce_UnexpectedValueException,
+			0,
+			"Expected %s::$%s value to be of type %s, but found %s",
+			ZSTR_VAL(php_ce_UUID->name),
+			UUID_BYTES_PROP,
+			zend_get_type_by_const(IS_STRING),
+			zend_zval_type_name(bytes)
+		);
+		return;
+	}
+
+	if (Z_STRLEN_P(bytes) != PHP_UUID_LEN) {
+		zend_throw_exception_ex(
+			spl_ce_UnexpectedValueException,
+			0,
+			"Expected %s::$%s value to be exactly %u bytes long, but found %u",
+			ZSTR_VAL(php_ce_UUID->name),
+			UUID_BYTES_PROP,
+			PHP_UUID_LEN,
+			Z_STRLEN_P(bytes)
+		);
+		return;
+	}
```

Comment:
Shouldn't this reuse the get_uuid() code? Also, I find it weird that this does the same checks as get_uuid(), but throws different exception types. And also ones that are not used elsewhere.